pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -42,6 +57,360 @@ type Adapter struct {
 	devfileRunCmd   string
 }
 
+const dockerfilePath string = "Dockerfile"
+
+func (a Adapter) runBuildConfig(client *occlient.Client, parameters common.BuildParameters) (err error) {
+	buildName := a.ComponentName
+
+	commonObjectMeta := metav1.ObjectMeta{
+		Name: buildName,
+	}
+
+	_, err = client.CreateDockerBuildConfigWithBinaryInput(commonObjectMeta, dockerfilePath, parameters.Tag, []corev1.EnvVar{})
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		// This will delete both the BuildConfig and any builds using that BuildConfig
+		derr := client.DeleteBuildConfig(commonObjectMeta)
+		if err == nil {
+			err = derr
+		}
+	}()
+
+	syncAdapter := sync.New(a.AdapterContext, &a.Client)
+	reader, err := syncAdapter.SyncFilesBuild(parameters, dockerfilePath)
+	if err != nil {
+		return err
+	}
+
+	bc, err := client.RunBuildConfigWithBinaryInput(buildName, reader)
+	if err != nil {
+		return err
+	}
+
+	reader, writer := io.Pipe()
+	s := log.Spinner("Waiting for build to finish")
+
+	var cmdOutput string
+	// This Go routine will automatically pipe the output from WaitForBuildToFinish to
+	// our logger.
+	go func() {
+		scanner := bufio.NewScanner(reader)
+		for scanner.Scan() {
+			line := scanner.Text()
+
+			if log.IsDebug() {
+				_, err := fmt.Fprintln(os.Stdout, line)
+				if err != nil {
+					log.Errorf("Unable to print to stdout: %v", err)
+				}
+			}
+
+			cmdOutput += fmt.Sprintln(line)
+		}
+	}()
+
+	if err := client.WaitForBuildToFinish(bc.Name, writer); err != nil {
+		s.End(false)
+		return errors.Wrapf(err, "unable to build image using BuildConfig %s, error: %s", buildName, cmdOutput)
+	}
+
+	s.End(true)
+	return
+}
+
+// Build image for devfile project
+func (a Adapter) Build(parameters common.BuildParameters) (err error) {
+	client, err := occlient.New()
+	if err != nil {
+		return err
+	}
+
+	isBuildConfigSupported, err := client.IsBuildConfigSupported()
+	if err != nil {
+		return err
+	}
+
+	if isBuildConfigSupported {
+		return a.runBuildConfig(client, parameters)
+	}
+
+	return errors.New("unable to build image, only Openshift BuildConfig build is supported")
+}
+
+func determinePort(envSpecificInfo envinfo.EnvSpecificInfo) string {
+	// TODO: Determine port to use (from env.yaml or other location!!)
```

Comment:
@groeges could we remove this TODO?

Reply:
Yes, as we have determined that we are initially only supporting 1 port and that will be the first non-Docker route in env.yaml, then I think this is done and therefore we can remove the TODO.